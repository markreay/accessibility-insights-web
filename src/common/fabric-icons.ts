// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { registerIcons } from '@uifabric/styling';

export function initializeFabricIcons(): void {
    registerIcons({
        style: {
            MozOsxFontSmoothing: 'grayscale',
            WebkitFontSmoothing: 'antialiased',
            fontStyle: 'normal',
            fontWeight: 'normal',
            speak: 'none',
        },
        fontFace: {
            fontFamily: 'FabricMDL2Icons',
        },
        icons: {
            add: '\uE710',
            back: '\uE72B',
            calculatorAddition: '\uE948',
            cancel: '\uE711',
            checkBox: '\uE739',
            checkMark: '\uE73E',
            chevronDown: '\uE70D',
            chevronDownMed: '\uE972',
            chevronRight: '\uE76C',
            chevronRightMed: '\uE974',
            chevronUp: '\uE70E',
            chromeClose: '\uE8BB',
            chromeMinimize: '\uE921',
            circleRing: '\uEA3A',
            completedSolid: '\uEC61',
            copy: '\uE8C8',
            contactCard: '\uEEBD',
            delete: '\uE74D',
            edit: '\uE70F',
            export: '\uEDE1',
            feedback: '\uED15',
            fileHTML: '\uF2ED',
            gear: '\uE713',
            giftboxOpen: '\uF133',
            globalNavButton: '\uE700',
            hide2: '\uEF89',
            home: '\uE80F',
            incidentTriangle: '\uE814',
            info: '\uE946',
            ladybugSolid: '\uF44A',
            mail: '\uE715',
            medical: '\uEAD4',
            play: '\uE768',
            refresh: '\uE72C',
            rocket: '\uF3B3',
            scopeTemplate: '\uF2B0',
            search: '\uE721',
            send: '\uE724',
            skypeCheck: '\uEF80',
            statusCircleCheckMark: '\uF13E',
            statusErrorFull: '\uEB90',
            stop: '\uE71A',
            tag: '\uE8EC',
            testBeaker: '\uF3A5',
            testBeakerSolid: '\uF3A6',
            undo: '\uE7A7',
            unknown: '\uE9CE',
            view: '\uE890',
        },
    });
}
