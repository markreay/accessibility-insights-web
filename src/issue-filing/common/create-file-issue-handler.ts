// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
import { EnvironmentInfo } from 'common/environment-info-provider';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { IssueFilingServicePropertiesMap } from 'common/types/store-data/user-configuration-store';

import { IssueFilingUrlProvider } from '../types/issue-filing-service';

export const createFileIssueHandler = <Settings>(
    getUrl: IssueFilingUrlProvider<Settings>,
    getSettings: (data: IssueFilingServicePropertiesMap) => Settings,
) => {
    return async (
        browserAdapter: BrowserAdapter,
        servicePropertiesMap: IssueFilingServicePropertiesMap,
        issueData: CreateIssueDetailsTextData,
        environmentInfo: EnvironmentInfo,
    ): Promise<void> => {
        const serviceConfig = getSettings(servicePropertiesMap);

        const url = getUrl(serviceConfig, issueData, environmentInfo);
        await browserAdapter.createActiveTab(url);
    };
};
