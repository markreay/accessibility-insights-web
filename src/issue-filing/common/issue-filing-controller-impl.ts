// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseStore } from 'common/base-store';
import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
import { EnvironmentInfo } from 'common/environment-info-provider';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';

import { IssueFilingServiceProvider } from '../issue-filing-service-provider';

export type IssueFilingController = {
    fileIssue: (serviceKey: string, issueData: CreateIssueDetailsTextData) => void;
};

export class IssueFilingControllerImpl implements IssueFilingController {
    constructor(
        private readonly provider: IssueFilingServiceProvider,
        private readonly browserAdapter: BrowserAdapter,
        private readonly environmentInfo: EnvironmentInfo,
        private readonly userConfigurationStore: BaseStore<UserConfigurationStoreData>,
    ) {}

    public fileIssue = (serviceKey: string, issueData: CreateIssueDetailsTextData): Promise<void> => {
        const service = this.provider.forKey(serviceKey);
        const userConfigurationStoreData = this.userConfigurationStore.getState();

        return service.fileIssue(this.browserAdapter, userConfigurationStoreData.bugServicePropertiesMap, issueData, this.environmentInfo);
    };
}
