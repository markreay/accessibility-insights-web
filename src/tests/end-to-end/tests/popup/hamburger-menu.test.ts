// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Browser } from '../../common/browser';
import { launchBrowser } from '../../common/browser-factory';
import { popupPageElementIdentifiers } from '../../common/element-identifiers/popup-page-element-identifiers';
import { formatPageElementForSnapshot } from '../../common/element-snapshot-formatter';
import { PopupPage } from '../../common/page-controllers/popup-page';
import { TargetPage } from '../../common/page-controllers/target-page';
import { scanForAccessibilityIssues } from '../../common/scan-for-accessibility-issues';

describe('Popup -> Hamburger menu', () => {
    let browser: Browser;
    let targetPage: TargetPage;
    let popupPage: PopupPage;

    beforeAll(async () => {
        browser = await launchBrowser({ suppressFirstTimeDialog: true });
        targetPage = await browser.newTargetPage();
        popupPage = await browser.newPopupPage(targetPage);
        await popupPage.clickSelector(popupPageElementIdentifiers.hamburgerMenuButton);
    });

    afterAll(async () => {
        if (browser) {
            await browser.close();
            browser = undefined;
        }
    });

    it('should have content matching snapshot', async () => {
        const hamburgerMenu = await formatPageElementForSnapshot(popupPage, popupPageElementIdentifiers.hamburgerMenu);
        expect(hamburgerMenu).toMatchSnapshot();
    });

    it.each([true, false])('should pass accessibility validation with highContrastMode=%s', async highContrastMode => {
        await browser.setHighContrastMode(highContrastMode);
        await popupPage.waitForHighContrastMode(highContrastMode);

        const results = await scanForAccessibilityIssues(popupPage, popupPageElementIdentifiers.hamburgerMenu);
        expect(results).toHaveLength(0);
    });
});
