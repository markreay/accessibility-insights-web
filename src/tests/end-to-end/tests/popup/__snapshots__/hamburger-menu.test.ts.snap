// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup -> Hamburger menu should have content matching snapshot 1`] = `
<DocumentFragment>
  <div
    class="ms-FocusZone css-000 ms-ContextualMenu is-open popup-menu root-000"
    data-focuszone-id="FocusZone000"
    role="presentation"
  >
    <ul
      class="ms-ContextualMenu-list is-open list-000"
      role="menu"
    >
      <li
        class="ms-ContextualMenu-item item-000"
        role="presentation"
      >
        <button
          aria-disabled="false"
          aria-posinset="1"
          aria-setsize="5"
          class="ms-ContextualMenu-link root-000"
          name="FastPass"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="ms-ContextualMenu-linkContent linkContent-000"
          >
            <i
              aria-hidden="true"
              class="ms-ContextualMenu-icon icon-000"
              data-icon-name="Rocket"
              role="presentation"
            >
              
            </i>
            <span
              class="ms-ContextualMenu-itemText label-000"
            >
              FastPass
            </span>
          </div>
        </button>
      </li>
      <li
        class="ms-ContextualMenu-item item-000"
        role="presentation"
      >
        <button
          aria-disabled="false"
          aria-posinset="2"
          aria-setsize="5"
          class="ms-ContextualMenu-link root-000"
          name="Assessment"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="ms-ContextualMenu-linkContent linkContent-000"
          >
            <i
              aria-hidden="true"
              class="ms-ContextualMenu-icon icon-000"
              data-icon-name="testBeakerSolid"
              role="presentation"
            >
              
            </i>
            <span
              class="ms-ContextualMenu-itemText label-000"
            >
              Assessment
            </span>
          </div>
        </button>
      </li>
      <li
        class="ms-ContextualMenu-item item-000"
        role="presentation"
      >
        <button
          aria-disabled="false"
          aria-posinset="3"
          aria-setsize="5"
          class="ms-ContextualMenu-link root-000"
          name="Ad hoc tools"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="ms-ContextualMenu-linkContent linkContent-000"
          >
            <i
              aria-hidden="true"
              class="ms-ContextualMenu-icon icon-000"
              data-icon-name="Medical"
              role="presentation"
            >
              
            </i>
            <span
              class="ms-ContextualMenu-itemText label-000"
            >
              Ad hoc tools
            </span>
          </div>
        </button>
      </li>
      <li
        aria-hidden="true"
        class="ms-ContextualMenu-divider divider-000"
        role="separator"
      />
      <li
        class="ms-ContextualMenu-item item-000"
        role="presentation"
      >
        <button
          aria-disabled="false"
          aria-posinset="4"
          aria-setsize="5"
          class="ms-ContextualMenu-link root-000"
          name="Keyboard shortcuts"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="ms-ContextualMenu-linkContent linkContent-000"
          >
            <i
              aria-hidden="true"
              class="ms-ContextualMenu-icon icon-000"
              role="presentation"
            />
            <span
              class="ms-ContextualMenu-itemText label-000"
            >
              Keyboard shortcuts
            </span>
          </div>
        </button>
      </li>
      <li
        class="ms-ContextualMenu-item item-000"
        role="presentation"
      >
        <button
          aria-disabled="false"
          aria-posinset="5"
          aria-setsize="5"
          class="ms-ContextualMenu-link root-000"
          name="Help"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="ms-ContextualMenu-linkContent linkContent-000"
          >
            <i
              aria-hidden="true"
              class="ms-ContextualMenu-icon icon-000"
              data-icon-name="Unknown"
              role="presentation"
            >
              
            </i>
            <span
              class="ms-ContextualMenu-itemText label-000"
            >
              Help
            </span>
          </div>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
